domain.go

```diff
@@ -1832,10 +1832,11 @@ func (d *Domain) ListAllInterfaceAddresses(src uint) ([]DomainInterface, error)
 
 		for k := 0; k < numAddr; k++ {
 			ifaces[i].Addrs[k] = DomainIPAddress{}
-			ifaces[i].Addrs[k].Type = int(addrSlice[k]._type)
-			ifaces[i].Addrs[k].Addr = C.GoString(addrSlice[k].addr)
-			ifaces[i].Addrs[k].Prefix = uint(addrSlice[k].prefix)
-
+			if addSlice[k] != nil {
```

Comment:
I don't see how this is possible to be NULL. This variable gets initialized from the 'addrs' field in 
virDomainInterface and that is an array of structs, not an array of pointers to structs, so addrSlice[k] is guaranteed non-NULL